### PR TITLE
Revamp group IDs of jsr94, janino and upgrade jxl, dools, Joda-time for BRS

### PR DESCRIPTION
**Pull request by milindaperera**

### CODE REVIEW

File: drools/6.3.0-wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            <name>Apache Snapshot Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/snapshots/</url>
+        </snapshotRepository>
+    </distributionManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.drools</groupId>
+            <artifactId>drools-jsr94</artifactId>
+            <version>${drool.version}</version>
```
Comment (arunasujith) [inline drools/6.3.0-wso2v1/pom.xml]: Please make all dependencies optional

File: drools/6.3.0-wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                        <Export-Package>
+                            org.drools.*;org.kie.*;version="${project.version}";-split-package:=merge-first
+                        </Export-Package>
+                        <Private-Package></Private-Package>
+                        <Import-Package>
+                            !org.drools.*;version="[6.3.0,6.4.0)",
+                            !org.kie.*;version="[6.3.0,6.4.0)"
+                        </Import-Package>
+                        <Embed-Dependency>
+                            drools-jsr94;knowledge-api;drools-core;drools-compiler;kie-api;kie-internal;scope=compile|runtime;inline=true;
```
Comment (arunasujith) [inline drools/6.3.0-wso2v1/pom.xml]: Can you explain the reason to add the embed-dependency

File: janino/2.5.15.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.codehaus.janino.*;version="${project.version}",
+                        </Export-Package>
+                        <Private-Package></Private-Package>
+                        <Import-Package>
+                            !org.codehaus.janino.*,
+                            org.apache.tools.ant.types,
```
Comment (arunasujith) [inline janino/2.5.15.wso2v1/pom.xml]: Define the import vesions

File: janino/2.5.15.wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                            org.apache.tools.ant.taskdefs,
+                            org.apache.tools.ant.taskdefs.compilers
+                        </Import-Package>                        
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (arunasujith) [inline janino/2.5.15.wso2v1/pom.xml]: Why there is dynamic imports?

File: joda-time/2.8.2.wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            <name>Apache Snapshot Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/snapshots/</url>
+        </snapshotRepository>
+    </distributionManagement>
+
+
+    <dependencies>
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
```
Comment (arunasujith) [inline joda-time/2.8.2.wso2v1/pom.xml]: Make dependency optional

File: joda-time/2.8.2.wso2v1/pom.xml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.joda.time.*;version="${project.version}",
+                        </Export-Package>
+                        <Private-Package></Private-Package>
+                        <Import-Package>
+                            org.joda.convert;resolution:=optional
+                        </Import-Package>
+                        <Embed-Dependency>joda-time;scope=compile|runtime;inline=true;</Embed-Dependency>
```
Comment (arunasujith) [inline joda-time/2.8.2.wso2v1/pom.xml]: Can you explain the reason to add the embed-dependency

File: jxl/2.6.12.wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                        <Private-Package></Private-Package>
+                        <Include-Resource>
+                            @jxl-${version.jxl}.jar!/functions.properties,
+                            @jxl-${version.jxl}.jar!/functions_da.properties,
+                            @jxl-${version.jxl}.jar!/functions_de.properties,
+                            @jxl-${version.jxl}.jar!/functions_en.properties,
+                            @jxl-${version.jxl}.jar!/functions_es.properties,
+                            @jxl-${version.jxl}.jar!/functions_fr.properties,
+                        </Include-Resource>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (arunasujith) [inline jxl/2.6.12.wso2v1/pom.xml]: Can you explain the reason to add dynamic imports

File: drools/6.3.0-wso2v1/pom.xml (hunk 1 above)
Comment (milindaperera) [inline drools/6.3.0-wso2v1/pom.xml]: dependency optional added

File: drools/6.3.0-wso2v1/pom.xml (hunk 2 above)
Comment (milindaperera) [inline drools/6.3.0-wso2v1/pom.xml]: Refer mail : [Dev] [BRS] Upgrading the Drools and its Dependencies

File: janino/2.5.15.wso2v1/pom.xml (hunk 4 above)
Comment (milindaperera) [inline janino/2.5.15.wso2v1/pom.xml]: Dynamic imports are required due to dynamic class loading is required for rules

File: joda-time/2.8.2.wso2v1/pom.xml (hunk 5 above)
Comment (milindaperera) [inline joda-time/2.8.2.wso2v1/pom.xml]: updated

File: jxl/2.6.12.wso2v1/pom.xml (hunk 7 above)
Comment (milindaperera) [inline jxl/2.6.12.wso2v1/pom.xml]: Dynamic imports are required due to dynamic class loading is required for rules

File: joda-time/2.8.2.wso2v1/pom.xml (hunk 6 above)
Comment (milindaperera) [inline joda-time/2.8.2.wso2v1/pom.xml]: Updated

**Pull request merged**